uwrt_mars_rover_utils/include/uwrt_mars_rover_utils/uwrt_can.h

```diff
@@ -131,6 +140,59 @@ class UWRTCANWrapper {
 
     return bytes_sent == sizeof(struct can_frame);
   }
+
+  template <class T>
+  bool writeToIDwithAck(T data, uint32_t id) {
+    // check socket has been initialized
+    if (!initialized_) {
+      ROS_ERROR_STREAM_NAMED(uwrt_mars_rover_utils::getLoggerName(), "Please intialize CAN wrapper before using it");
```

Comment:
Rather than calling getloggername each time, you should call it once in the constructor of uwrt can and save it in a member variable(typically just called name_).

Also could u also fix all the other prints in this file to be named